govcd/catalog.go

```diff
@@ -884,7 +899,22 @@ func (cat *Catalog) GetCatalogItemById(catalogItemId string, refresh bool) (*Cat
 	return nil, ErrorEntityNotFound
 }
 
-// GetCatalogItemByNameOrId finds a Catalog Item by Name or ID
+// GetVAppTemplateById finds a vApp Template by ID.
+// On success, returns a pointer to the VAppTemplate structure and a nil error.
+// On failure, returns a nil pointer and an error.
+func (cat *Catalog) GetVAppTemplateById(vAppTemplateId string) (*VAppTemplate, error) {
+	vappTemplateHref := cat.client.VCDHREF
+	vappTemplateHref.Path += "/vAppTemplate/vappTemplate-" + strings.ReplaceAll(vAppTemplateId, "urn:vcloud:vapptemplate:", "")
```

Comment:
You could use `extractUuid` on `vAppTemplateId` here and in `Delete` function. 

Because of this mess with UUIDs/URNs in VCD, we have a bunch of functions here in SDK to deal with that. Some are `getBareEntityUuid`, `extractUuid`, `equalIds`, `IsUuid`, `isUrn`

Reply:
I finally used `getBareEntityUuid` as in this case I needed the inner UUID, not the full URN.
Thanks for the suggestion, I've noted them for subsequent changes 🙂